awx/main/models/events.py

```diff
@@ -535,16 +536,19 @@ def _update_host_summary_from_stats(self, hostnames):
             # in single queries
             host_mapping = dict((summary['host_id'], summary['id']) for summary in JobHostSummary.objects.filter(job_id=job.id).values('id', 'host_id'))
             updated_hosts = set()
+            updated_hosts_list = []
             for h in all_hosts:
                 # if the hostname *shows up* in the playbook_on_stats event
                 if h.name in hostnames:
                     h.last_job_id = job.id
                     updated_hosts.add(h)
+                    updated_hosts.append(h.name)
                 if h.id in host_mapping:
                     h.last_job_host_summary_id = host_mapping[h.id]
                     updated_hosts.add(h)
```

Comment:
I am not sure if i should make another list and append ```h.id``` is that. I only need ```h.name``` ( hostname ) in my new table/model. 
Let me know.

Reply:
I don't believe you should be managing a new list or lists at all here.  `updated_hosts` is already a set that is getting the full `Host` objects.  In your line 551 below, you could and should be constructing your list comprehension off of that.